python/dgl/_ffi/base.py

```diff
@@ -148,3 +148,38 @@ def load_tensor_adapter(backend, version):
     if not tensor_adapter_loaded:
         logger = logging.getLogger("dgl-core")
         logger.debug("Memory optimization with PyTorch is not enabled.")
+
+dgl_sparse_loaded = False
```

Comment:
* remove this global variable.
* move the entire `load_dgl_sparse` logic to `dgl/mock_sparse2`.